### PR TITLE
First commit for yaml-cpp conda-forge package

### PR DESCRIPTION
**Pull request by allanleal**
This PR is an effort to add yaml-cpp library to conda-forge, which fixes issue #4522.

### CODE REVIEW

File: recipes/yaml-cpp/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  name: yaml-cpp
+  version: {{ version }}
+
+source:
+  url: https://github.com/jbeder/yaml-cpp/archive/yaml-cpp-{{ version }}.tar.gz
+  sha256: e4d8560e163c3d875fd5d9e5542b5fd5bec810febdcba61481fe5fc4e6b1fd05
+
+build:
+  number: 0
+  # skip: True  # [win and vc<15]
```
Comment (marcelotrevisani) [inline recipes/yaml-cpp/meta.yaml]: please, remove this comment

**Comment by allanleal** [comment]
Hi @marcelotrevisani - thanks a lot for helping on this PR! :+1:

**Comment by allanleal** [comment]
@marcelotrevisani We still need to figure out what is going on for the win build in AppVeyor. The error message is not clear:

~~~
-- Build files have been written to: C:/bld/yaml-cpp_1551406276538/test_tmp/test/build
[1/2] Building CXX object CMakeFiles\test.dir\main.cpp.obj
cl : Command line warning D9025 : overriding '/MD' with '/MDd'
[2/2] Linking CXX executable test.exe
main.cpp.obj : MSIL .netmodule or module compiled with /GL found; restarting link with /LTCG; add /LTCG to the link command line to improve linker performance
LINK : warning LNK4075: ignoring '/INCREMENTAL' due to '/LTCG' specification
Generating code
Finished generating code
main.cpp.obj : MSIL .netmodule or module compiled with /GL found; restarting link with /LTCG; add /LTCG to the link command line to improve linker performance
LINK : warning LNK4075: ignoring '/INCREMENTAL' due to '/LTCG' specification
Generating code
Finished generating code
Tests failed for yaml-cpp-0.6.2-he025d50_0.tar.bz2 - moving package to C:\bld\broken
WARNING:conda_build.build:Tests failed for yaml-cpp-0.6.2-he025d50_0.tar.bz2 - moving package to C:\bld\broken
TESTS FAILED: yaml-cpp-0.6.2-he025d50_0.tar.bz2
Command exited with code 1
~~~

**Comment by allanleal** [comment]
@chrisburr The test is passing for the win build (which we now know after your printing messages). However, there is a failure after this that I'm not sure what the message means.

~~~
Generating code
Finished generating code
"Ninja finished"
"Ninja okay"
"Test finished"
"Test okay"
Tests failed for yaml-cpp-0.6.2-he025d50_0.tar.bz2 - moving package to C:\bld\broken
WARNING:conda_build.build:Tests failed for yaml-cpp-0.6.2-he025d50_0.tar.bz2 - moving package to C:\bld\broken
TESTS FAILED: yaml-cpp-0.6.2-he025d50_0.tar.bz2
Command exited with code 1
~~~

**Comment by chrisburr** [comment]
Unfortunately I have no idea why it thinks it failed.

**Comment by tadeu** [comment]
> Unfortunately I have no idea why it thinks it failed.

The test was being built in Debug, and it failed in runtime with a negative status code

**Comment by allanleal** [comment]
Excellent - thanks for finding the issue @tadeu !

**Comment by marcelotrevisani** [comment]
> > Unfortunately I have no idea why it thinks it failed.
> 
> The test was being built in Debug (the default), and it failed in runtime with a negative status code (since the package is in Release)

@tadeu saving the day! 🎉 
I didn't realize that, thanks!

**Comment by marcelotrevisani** [comment]
One thing which I fixed before was: ``CMAKE`` was not finding the headers in ``windows`` platform.

**Comment by allanleal** [comment]
@marcelotrevisani and @tadeu - can I add you both to the list of maintainers as an acknowledgement of your substantial help here?

**Comment by marcelotrevisani** [comment]
Just one detail here... according to the documentation this test file should print
```
Here's the output YAML:
Hello, World!
```

but it is printing just:
```
Here's the output YAML:
```

**Comment by marcelotrevisani** [comment]
> @marcelotrevisani and @tadeu - can I add you both to the list of maintainers as an acknowledgement of your substantial help here?

Sure! :)

**Comment by allanleal** [comment]
Good catch! Maybe this example is not valid in this version of yaml-cpp, 
and the docs did not change? I'll try a different example in my machine 
and change the one in the recipe.

**Comment by allanleal** [comment]
@marcelotrevisani , @tadeu - thanks a lot!

**Comment by allanleal** [comment]
@marcelotrevisani you may want to close issue #4522 (which requested this package).

**Pull request merged**